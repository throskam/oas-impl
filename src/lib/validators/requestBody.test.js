import RequestBodyValidator from './requestBody'

describe('Empty content', () => {
  const requestBody = {
    required: true
  }

  const validator = new RequestBodyValidator(requestBody)

  it('should no rerros', () => {
    const payload = { value: 1, mediaType: 'application/json' }
    expect(validator.validate(payload)).toMatchSnapshot()
  })
})

describe('Required', () => {
  const requestBody = {
    required: true,
    content: {
      'application/json': {
        schema: {
          type: 'integer'
        }
      }
    }
  }

  const validator = new RequestBodyValidator(requestBody)

  it('should return no errors when the value is defined', () => {
    const payload = { value: 1, mediaType: 'application/json' }
    expect(validator.validate(payload)).toMatchSnapshot()
  })

  it('should return a required error when the value is missing', () => {
    expect(validator.validate()).toMatchSnapshot()
  })
})

describe('Optional', () => {
  const requestBody = {
    content: {
      'application/json': {
        schema: {
          type: 'integer'
        }
      }
    }
  }

  const validator = new RequestBodyValidator(requestBody)

  it('should return no errors when the value is defined', () => {
    const payload = { value: 1, mediaType: 'application/json' }
    expect(validator.validate(payload)).toMatchSnapshot()
  })

  it('should return no errors when the value is missing', () => {
    expect(validator.validate()).toMatchSnapshot()
  })
})

describe('Content', () => {
  const requestBody = {
    content: {
      'application/json': {
        schema: {
          type: 'integer'
        }
      }
    }
  }

  const validator = new RequestBodyValidator(requestBody)

  it('should return a validation error when the value is invalid', () => {
    const payload = { value: 'string', mediaType: 'application/json' }
    expect(validator.validate(payload)).toMatchSnapshot()
  })
})
