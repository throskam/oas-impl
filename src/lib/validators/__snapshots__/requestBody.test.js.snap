// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "requestBody.content[application/json].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Empty content should no rerros 1`] = `Array []`;

exports[`Optional should return no errors when the value is defined 1`] = `Array []`;

exports[`Optional should return no errors when the value is missing 1`] = `Array []`;

exports[`Required should return a required error when the value is missing 1`] = `
Array [
  Object {
    "path": "requestBody",
    "rule": "requestBody-required",
  },
]
`;

exports[`Required should return no errors when the value is defined 1`] = `Array []`;
